source/components/ResolutionSlider.js

```diff
@@ -1,43 +1,111 @@
 import React from 'react'
-import {scaleLinear} from 'd3-scale'
-
-import {tileEdgeRange} from '../constants'
-
-const normalizeValue = scaleLinear()
-  .domain([tileEdgeRange.min, tileEdgeRange.max])
-  .range([0, 100])
+import {scaleLog} from 'd3-scale'
 
 export default class ResolutionSlider extends React.Component {
   constructor(props) {
     super(props)
-
     this.state = {
-      value: normalizeValue(tileEdgeRange.default),
+      value: 50,
+      typedValue: '',
+    }
+    this.normalizeValue = scaleLog().domain(this.props.metricDomain).range([1, 99])
+    this._triggerChangeFromText = this._triggerChangeFromText.bind(this)
+  }
+
+  componentDidMount() {
+    this._triggerChange(50, {normalized: true}) // reset slider to 50
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (JSON.stringify(nextProps) !== JSON.stringify(this.props)) {
+      this.normalizeValue.domain(nextProps.metricDomain) // update the domain
+      this._triggerChange(50, {normalized: true})
+    }
+  }
+
+  _toSignificant(number) {
+    return parseFloat(number.toPrecision(2))
+  }
+
+  _triggerChange(value, options) {
+    /**
+     * changed can be triggered either
+     * 1) with a normalized value (in the range of this._normalizeValue) or
+     * 2) with a denormalized value (in the domain of this._normalizeValue)
+     * the first case is primarily for the slider
+     * the second case is for the typed box
+     * note: the value displayed in the input box is always the value passed
+     * to the props.onChange handler so that it is the source of truth
+     * for the hexagon metrics
+     */
+    options = options || {normalized: false}
+    if (options.normalized) {
+      const deNormalizedValue = this._toSignificant(this.normalizeValue.invert(value))
+      this.setState({value, typedValue: deNormalizedValue})
+      this.props.onChange(deNormalizedValue)
+    } else {
+      const normalizedValue = this._toSignificant(this.normalizeValue(value))
+      this.setState({value: normalizedValue, typedValue: value})
+      this.props.onChange(value)
```

Comment:
Can we eliminate the conditional in this function?

If only the event from the text input may be out of bounds, couldn't `_triggerChangeFromText` take responsibility for that sanitization before handing off the cleaned-up value to this common exit point?